### PR TITLE
Add tests to verify that most recent column change is considered when generating burndown chart

### PR DESCRIPTION
**Pull request by wmde-manicki**
Task: https://phabricator.wikimedia.org/T103071

As I understand the problem described in Phabricator ticket, at some point Phragile considered tasked being closed at the point it first got moved to the "closed" column, even if it has been moved back to "not closed" column afterwards.
If I am not mistaken this behaviour has been (possibly unintentionally) fixed along some other changes at some point (didn't manage to find the actual change).
`BurndownChartTest::testClosedPerDayConsidersMostRecentColumnChangeInWorkboardMode` is about checking the behaviour I described above. Also adding tests for two other classes involved that prove that the most recent column-change transaction is considered.

If I am wrong in what the Phab task is about, please comment!

### CODE REVIEW

File: tests/unit/StatusByWorkboardDispatcherTest.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	{
+		$phabricatorAPI = $this->getMockBuilder( PhabricatorAPI::class )
+			->disableOriginalConstructor()
+			->getMock();
+		$phabricatorAPI->method('queryPHIDs')->will($this->returnCallback(function()
+		{
+			return array_map(function($column)
+			{
+				return ['name' => $column];
+			}, $this->workboardColumns);
```
Comment (WMDE-Fisch) [inline tests/unit/StatusByWorkboardDispatcherTest.php]: you could also use array_flip( $this->columnPHIDs ) and get rid of the field

**Comment by WMDE-Fisch** [comment]
+ 1,99

File: tests/unit/StatusByWorkboardDispatcherTest.php (hunk 1 above)
Comment (wmde-manicki) [inline tests/unit/StatusByWorkboardDispatcherTest.php]: True, I had that before like:
```php
return array_map(function($column)
{
	return ['name' => $column];
}, array_flip($this->workboardColumns));
```
but looking at this made me look like Jeroen's "double space OMG" cat.
But it doesn't really read that bad, so I'd probably change it the way you suggested.

**Comment by wmde-manicki** [comment]
So I adjusted a bit `StatusByWorkboardDispatcher`'s tests to address @WMDE-Fisch's suggestion, plus split a length test method into three separate methods so tests stick better to Arrange-Act-Assert.

**Pull request merged**